### DISCUSSION TITLE
Need to send custom http headers in 304 responses

### DISCUSSION THREAD

**Discussion by JaconsMorais** [vercel/community#792] [Help]
Hi, I have an app running on Vercel using NextJS v10.0.7.

When requesting a page using "If-None-Match": <etag> header, Vercel returns 304 response with invalid header for cache.

When acessing without if-none-match header, I got every header I have setted in my vercel.json file.

Here is my vercel.json file
``
{
  "routes": [
    {
      "src": "/quem-somos/",
      "headers": {
        "cdn-cache-control": "public, max-age=31536000, immutable",
        "cache-control": "public, max-age=31536000, immutable"
      }
    }
  ]
}
``

Response headers without if-none-match header and response status 200

![imagem_2022-08-22_152505488](https://user-images.githubusercontent.com/6156243/186002345-7cd53306-3d6c-4c3f-82a9-b304ecfe0fda.png)

Response headers with if-none-match with valid etag and response status 304

![image](https://user-images.githubusercontent.com/6156243/186002379-68653bcd-b984-4405-9d0e-689b5808e7e2.png)

I'm using cloudflare on top of my structure, so I have to force Vercel to respond correct headers even in 304 responses.

I just need to have cdn-cache-control in 304 responses.

What I'm missing? What should I do?

**Comment by amyegan** [comment]
Can you explain a little more about what you mean by this?
> I'm using cloudflare on top of my structure, so I have to force Vercel to respond correct headers even in 304 responses.

I would recommend taking a look at our guide on [using Cloudflare with Vercel](https://vercel.com/guides/using-cloudflare-with-vercel#with-proxy) to see if that setup might simplify things for you.

If you can share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), then I can dig into the problem more.

**Reply by JaconsMorais** [reply]
Hi @amyegan, thank you for your repply.

Let me explain more clearly how it works.

Cloudflare send an etag in every request made to my app(in Vercel network), when Vercel check that etag and its valid, he return a 304 response to Cloudflare servers.

The problem is: When Vercel made this response, he change all headers, sending cache-control: no-cache, no-store or not even sending my custom header cdn-cache-control.

And when Cloudflare receive this invalid cache header, they change TTL of my assets to 7200 seconds, and this is a problem to me.

So, I need to get cache-control and cdn-cache-control headers like I get in status 200 responses.

**Reply by JaconsMorais** [reply]
The screenshot I have made I could change cache-control to another response, but I really need to get cdn-cache-control too.

**Reply by crowterli** [reply]
Hi @JaconsMorais. Could you please raise a case for this at vercel.com/support/request and share with us the ticket number so I can dig deeper into this. Thanks.

**Reply by JaconsMorais** [reply]
Hi @crowterli, thank you for your response.
I submitted a ticket, the number is #00095437
